rama-http-types/src/curl/traits.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{Body, Request, Parts}
+use crate::dep::http_body;
+use bytes::Bytes;
+use rama_core::error::{BoxError, OpaqueError};
+
+mod sealed {
+    pub trait Sealed {}
+
+    impl<B> Sealed for crate::Request<B> 
+    where 
+    B: http_body::Body<Data = Bytes, Error = Into<BoxError>> + Send + Sync 
+    {}
+
+    impl Sealed for crate::Parts {}
+
+    impl<T: Sealed> Sealed for &T {}
+}
+
+/// Sealed trait for converting HTTP request headers into a curl command
+
+pub trait IntoCurlHeadersCommand: sealed::Sealed {
+    fn into_curl_headers_command(&self) -> Result<String, OpaqueError>;
```

Comment:
`into_curl_command` is indeed gonna be fallible as you'll need to consume the body, but pretty certain that `into_curl_headers_command` can be infallible, so can just return a `String`. Accessing header data cannot fail I think. Not in a way that is anyway recoverable.

Reply:
Also `IntoCurlHeadersCommandPriv` probably should just provide some methods to give access to the relevant parts. `Http1HeaderMap` (or however it's called), http version, method, uri, ... Where all except the h1HeaderMap are returned as reference